### PR TITLE
Relocating contributing guidelines to docs site 

### PR DESCRIPTION
**Pull request by pmcmaw**
Relocating contributing guidelines to a central location (docs site) rather than having copies in all the modules. This means every time we make an update it will only need to be done in 1 place rather than in all of our modules.

### CODE REVIEW

File: _docs/contributing_to_a_module.md [hunk 1]
```diff
@@ -0,0 +1,284 @@
...
+```ruby
+it 'does a thing' do
+  expect(subject).to contain_file("a test file").with({:path => "/etc/sample"})
+end
+```
+
+### Acceptance Tests
+
+A common pattern for acceptance tests is to create a test manifest, apply it
+twice to check for idempotency or errors, then run expectations.
```
Comment (DavidS) [inline _docs/contributing_to_a_module.md:237]: link to write a litmus test here? do we have that docs?

File: _docs/contributing_to_a_module.md [hunk 2]
```diff
@@ -0,0 +1,284 @@
...
+
+  idempotent_apply(pp)
+end
+
+describe file("/etc/sample") do
+  it { is_expected.to contain "test string" }
+end
+```
+
+# As as Trusted Contributor
```
Comment (DavidS) [inline _docs/contributing_to_a_module.md:257]: This section reads a bit cart-before-the-horse as there is no other mention of review policies in here. Maybe pull out the TC-relevant bit into a link to the actual Trusted Contributor docs and only keep the general review "four eyes" principle here?

File: _docs/contributing_to_a_module.md [hunk 3]
```diff
@@ -0,0 +1,284 @@
...
+
+Even if you have commit access to the repository, you still need to go through the process above, and have someone else review and merge
+in your changes.  The rule is that **all changes must be reviewed by a project developer that did not write the code to ensure that
+all changes go through a code review process.**
+
+The record of someone performing the merge is the record that they performed the code review. Again, this should be someone other than the author of the topic branch.
+
+# Getting your PR reviewed
```
Comment (DavidS) [inline _docs/contributing_to_a_module.md:265]: Could this use a link to the Community Day announcement/docs?

File: _docs/contributing_to_a_module.md [hunk 4]
```diff
@@ -0,0 +1,284 @@
...
+      may wish to make.
+
+## Testing
+## Getting Started
+
+Our Puppet modules provide [`Gemfile`](./Gemfile)s, which can tell a Ruby package manager such as [bundler](http://bundler.io/) what Ruby packages,
+or Gems, are required to build, develop, and test this software.
+
+Please make sure you have [bundler installed](http://bundler.io/#getting-started) on your system, and then use it to
+install all dependencies needed for this project in the project root by running
```
[the remark this answers is not in the document]
Comment (pmcmaw) [reply _docs/contributing_to_a_module.md:146]: By this, do you mean, update to use the experimental `pdk bundle` command or what exactly is it your ate suggesting? (Apologies, slightly lost here).

[the remark this answers is not in the document]
Comment (DavidS) [reply _docs/contributing_to_a_module.md:146]: yeah, exactly. Installing and using the PDK is something we support as a company. Even if the `pdk bundle` subcommand is experimental, there's a lot less to explain than "how to install ruby on your box" - which would be a pre-requisite for everything that's explained here.

File: _docs/contributing_to_a_module.md [hunk 5]
```diff
@@ -0,0 +1,287 @@
...
+
+## Getting your PR reviewed
+
+If you would like information about:
+
+- How to get your PR reviewed
+- What does a review consist of?
+- Who can merge my PR?
+
+The record of someone performing the merge is the record that they performed the code review. Again, this should be someone other than the author of the topic branch.
```
Comment (DavidS) [inline _docs/contributing_to_a_module.md:273]:
```suggestion
If there is no Github Code Review, the record of someone performing the merge is the record that they performed the code review.
Again, this should be someone other than the author of the topic branch.
```

**Comment by DavidS** [comment]
https://dev.to/codesandboxio/how-to-make-your-first-open-source-contribution-2oim (via https://twitter.com/ceeoreo_/status/1390342216485425155) is a nice getting started uide (and has more resources linked at the bottom). might be a good "what is OSS even?" starting place to link to, to pad out the content we don't want/need to write ourselves.

File: _docs/contributing_to_a_module.md [hunk 6]
```diff
@@ -0,0 +1,287 @@
...
+will allow you to run interactive debugging sessions using pry.
+
+## Running Tests on your local machine
+
+With all dependencies in place and up-to-date, run the tests:
+
+### Unit Tests
+
+```shell
+% pdk bundle exec rake spec
```
Comment (binford2k) [inline _docs/contributing_to_a_module.md:201]: It might be worth a quick note explaining why we use this instead of `pdk test unit`

**Comment by pmcmaw** [comment]
Thank you for your feedback @binford2k and @DavidS :-) 
I have now addressed your comments.

File: _docs/contributing_to_a_module.md [hunk 7]
```diff
@@ -0,0 +1,295 @@
...
+  - PRs
+  - pull requests
+  - code review
+---
+
+So you want to contribute to a Puppet module: Great! Below are some instructions to get you started doing
+that very thing while setting expectations around code quality as well as a few tips for making the
+process as easy as possible.
+
+Although our reviews don't catch everything, this is final defense line for anything our automated tests cannot catch.
```
Comment (sanfrancrisko) [inline _docs/contributing_to_a_module.md:20]:
```suggestion
Although our reviews don't catch everything, this is the final defence line for anything our automated tests cannot catch.
```

File: _docs/contributing_to_a_module.md [hunk 8]
```diff
@@ -0,0 +1,295 @@
...
+      in order to open a pull request.
+
+  3.  Update the related JIRA issue.
+
+      If there is a JIRA issue associated with the change you
+      submitted, then you should update the ticket to include the
+      location of your branch, along with any other commentary you
+      may wish to make.
+
+## More indepth information on Contributing
```
Comment (sanfrancrisko) [inline _docs/contributing_to_a_module.md:139]:
```suggestion
## More in-depth information on Contributing
```

File: _docs/contributing_to_a_module.md [hunk 9]
```diff
@@ -0,0 +1,295 @@
...
+
+Thanks to [Ceora](https://twitter.com/ceeoreo_/status/1390342216485425155), she has put together a great article on ['How to Make Your First Open Source Contribution'](https://dev.to/codesandboxio/how-to-make-your-first-open-source-contribution-2oim). The information she has included is fanstatic and if you want more indepth information it is definitely the perfect place to start.
```
Comment (sanfrancrisko) [inline _docs/contributing_to_a_module.md:141]:
```suggestion
Thanks to [Ceora](https://twitter.com/ceeoreo_/status/1390342216485425155), she has put together a great article on ['How to Make Your First Open Source Contribution'](https://dev.to/codesandboxio/how-to-make-your-first-open-source-contribution-2oim). The information she has included is fantastic and if you want more in-depth information it is definitely the perfect place to start.
```

File: _docs/contributing_to_a_module.md [hunk 10]
```diff
@@ -0,0 +1,295 @@
...
+![automated_test_suite](test_run_info.png)
+
+- A green tick (Tests have passed)
+- A red cross (Tests have failed)
+- A yellow dot (Tests are still running)
+- A grey dot (The job has been cancelled)
+
+A PR will not be merged with failing tests, if you click on the small red cross icon it will give you additional information
+on what has actually failed. Once this is addressed you can commit your fix and continue working. If your fix is difficult and
+you need to do additional debugging check out the section in this document called **Running tests on your local machine** as this
```
Comment (sanfrancrisko) [inline _docs/contributing_to_a_module.md:195]: Could potentially link to it:
```suggestion
you need to do additional debugging check out the section in this document called [**Running tests on your local machine**](#running-tests-on-your-local-machine)  as this
```

File: _docs/contributing_to_a_module.md [hunk 11]
```diff
@@ -0,0 +1,295 @@
...
+
+### Unit Tests
+
+```shell
+% pdk bundle exec rake spec
+```
+
+This executes all the [rspec tests](http://rspec-puppet.com/) in the directories defined [here](https://github.com/puppetlabs/puppetlabs_spec_helper/blob/699d9fbca1d2489bff1736bb254bb7b7edb32c74/lib/puppetlabs_spec_helper/rake_tasks.rb#L17) and so on.
+rspec tests may have the same kind of dependencies as the module they are testing. Although the module defines these dependencies in its [metadata.json](./metadata.json),
+rspec tests define them in [.fixtures.yml](./fixtures.yml). We use this over `pdk test unit` as it provides live feedback as the tests are running. See below:
```
Comment (sanfrancrisko) [inline _docs/contributing_to_a_module.md:210]: Just curious about the `metadata.json` and `.fixtures.yml` links and what they're supposed to link to?

File: _docs/contributing_to_a_module.md [hunk 12]
```diff
@@ -0,0 +1,295 @@
...
+
+Thanks to [Ceora](https://twitter.com/ceeoreo_/status/1390342216485425155), she has put together a great article on ['How to Make Your First Open Source Contribution'](https://dev.to/codesandboxio/how-to-make-your-first-open-source-contribution-2oim). The information she has included is fanstatic and if you want more indepth information it is definitely the perfect place to start.
+
+# Testing
+## Getting Started
+
+Our Puppet modules provide a Gemfile in the module root directory, we can then use the pdk to install all of the Gems listed in the Gemfile.
+
+which can tell a Ruby package manager such as [bundler](http://bundler.io/) what Ruby packages,
+or Gems, are required to build, develop, and test this software.
```
Comment (sanfrancrisko) [inline _docs/contributing_to_a_module.md:149]: Is this supposed to be one sentence?

File: _docs/contributing_to_a_module.md [hunk 13]
```diff
@@ -0,0 +1,295 @@
...
+
+All Puppet Supported modules come with acceptance tests, which use [puppet litmus][puppet-litmus].
+Litmus supports multiple provisioners, that exist in the [provision module][provision-module].
+For information on how you can run your tests using litmus please check out the [litmus docs section][litmus-docs]
+
+## Writing Tests
+
+### Unit Tests
+
+When writing unit tests for Puppet, [rspec-puppet][] is your best friend. It provides tons of helper methods for testing your manifests against a
```
Comment (sanfrancrisko) [inline _docs/contributing_to_a_module.md:225]:
```suggestion
When writing unit tests for Puppet, [rspec-puppet][rspec-puppet] is your best friend. It provides tons of helper methods for testing your manifests against a
```

File: _docs/contributing_to_a_module.md [hunk 14]
```diff
@@ -0,0 +1,295 @@
...
+
+- Fork the module repository on GitHub and clone to your workspace
+
+- Make your changes!
+
+## Commit Checklist
+
+### The PR Title
+
+Our CHANGELOGs when releasing are automated and to do this, we make use of the PR title. This means that it is super important that your PR title always reflects the changes contained within the PR. Anything you think would be useful for an end user to know should be included.
```
Comment (sanfrancrisko) [inline _docs/contributing_to_a_module.md:32]:
```suggestion
The CHANGELOG in a module is generated automatically, using the PR title This means that it is super important that your PR title reflects the changes contained within the PR. Anything you think would be useful for an end user to know should be included.
```

**Review by sanfrancrisko — CHANGES_REQUESTED** [review]

File: _docs/contributing_to_a_module.md (hunk 11 above)
Comment (pmcmaw) [reply _docs/contributing_to_a_module.md:210]: Good shout, initially this was per repo but I have moved it out. I will work on updating this.

**Comment by pmcmaw** [comment]
I have combined and updated your comments @sanfrancrisko 
Apologies I was on PTO when I got this feedback.

**Review by sanfrancrisko — APPROVED** [review]

**Pull request merged**